dist/yoast-seo.js

```diff
@@ -334,7 +337,11 @@ YoastSEO.Analyzer.prototype.linkCount = function() {
 		for ( var i = 0; i < linkMatches.length; i++ ) {
 			var linkKeyword = this.linkKeyword( linkMatches[ i ] );
 			if ( linkKeyword ) {
-				linkCount.totalKeyword++;
+				if ( this.config.keyword !== "" ) {
```

Comment:
This is saved seperately so we can differentiate in the scoring, if no keyword is found the scoring is different.